File: Lab04/FragmentLambda_Java/app/src/main/java/com/example/fragmentlambda_java/RegisterFragment.java
package com.example.fragmentlambda_java;

import android.content.Context;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

public class RegisterFragment extends Fragment {

    private OnUserRegisteredListener mListener;

    public interface OnUserRegisteredListener {
        void onUserRegistered(User user);
    }

    public static RegisterFragment newInstance() {
        return new RegisterFragment();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnUserRegisteredListener) {
            mListener = (OnUserRegisteredListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnUserRegisteredListener");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_register, container, false);

        EditText edtUsername = view.findViewById(R.id.edtUsername);
        EditText edtPassword = view.findViewById(R.id.edtPassword);
        Button btnSubmit = view.findViewById(R.id.btnSubmit);

        btnSubmit.setOnClickListener(v -> {
            User user = new User(edtUsername.getText().toString(), edtPassword.getText().toString());
            mListener.onUserRegistered(user);
        });

        return view;
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }
}

File: Lab04/FragmentLambda/app/src/main/java/com/example/fragmentlambda/RegisterFragment.kt
package com.example.fragmentlambda

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText

class RegisterFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_register, container, false)

        val edtUsername = view.findViewById<EditText>(R.id.edtUsername)
        val edtPassword = view.findViewById<EditText>(R.id.edtPassword)
        val btnSubmit = view.findViewById<Button>(R.id.btnSubmit)

        btnSubmit.setOnClickListener {
            val user = User(
                username = edtUsername.text.toString(),
                password = edtPassword.text.toString()
            )
            onUserRegistered(user)
        }

        return view
    }

    companion object {
        private lateinit var onUserRegistered: (User) -> Unit

        @JvmStatic
        fun newInstance(onUserRegistered: (User) -> Unit): RegisterFragment {
            this.onUserRegistered = onUserRegistered
            return RegisterFragment()
        }
    }
}
